### DISCUSSION TITLE
Development docs

### DISCUSSION THREAD

**Discussion by CyberTailor** [mesonbuild/meson#11775] [General]
I couldn't find any development docs for Meson (for e.g. adding new compilers). Do they exist?

**Comment by eli-schwartz** [comment]
The current state of affairs is that this more or less requires you to figure it out yourself or ask questions.

A good starting point would be looking at how #8706 handled this when adding cython support. Basically the mesonbuild/compilers/ subsystem needs to have detect.py find a compiler and also add a new implementation file $lang.py which provides the necessary interface. Then backend/backend.py / ninjabackend.py should produce build edges when that language is used.
